src/ui/components/CollapsibleText/CollapsibleText.native.test.tsx

```diff
@@ -1,20 +1,106 @@
 import React from 'react'
 
-import { render, screen } from 'tests/utils'
-import { CollapsibleText } from 'ui/components/CollapsibleText/CollapsibleText'
+import { act, fireEvent, render, screen, userEvent } from 'tests/utils'
+
+import { CollapsibleText } from './CollapsibleText'
 
 const TEXT =
   'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas nec tellus in magna convallis egestas eget id justo. Donec lorem ante, tempor eu diam quis, laoreet rhoncus tortor. Sed posuere quis sapien sit amet rutrum. Nam arcu dui, blandit vitae massa ac, pulvinar rutrum tellus. Mauris molestie, sapien quis elementum interdum, ipsum turpis varius lorem, quis luctus tellus est et velit. Curabitur accumsan, enim ac tincidunt varius, lectus ligula elementum elit, a porta velit libero quis nunc. Maecenas semper augue justo, ac dapibus erat porttitor quis. Cras porttitor pharetra quam, et suscipit felis fringilla in. Aliquam ultricies mauris at vehicula finibus. Donec sed justo turpis. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Nunc dictum tempus velit, nec volutpat dolor fermentum non. Nullam efficitur diam nec orci aliquam, ut accumsan turpis convallis. Duis erat diam, ultricies non dolor a, elementum sagittis nibh. Curabitur dapibus ipsum eget quam scelerisque, eget venenatis urna laoreet.'
 const NUMBER_OF_LINES = 5
 
-jest.mock('libs/firebase/analytics/analytics')
+const mockOnLayoutWithButton = {
+  nativeEvent: {
+    layout: {
+      width: 375,
+      height: 2100,
+    },
+  },
+}
+const mockOnLayoutWithoutButton = {
+  nativeEvent: {
+    layout: {
+      width: 375,
+      height: 150,
+    },
+  },
+}
+const mockOnTextLayoutWhenEllipsis = {
+  nativeEvent: {
+    lines: [
+      { width: 100 },
+      { width: 100 },
+      { width: 100 },
+      { width: 100 },
+      { width: 370 },
+      { width: 100 },
+    ],
+  },
+}
+const mockOnTextLayoutWhenTooShort = {
+  nativeEvent: {
+    lines: [{ width: 100 }, { width: 100 }, { width: 100 }, { width: 100 }, { width: 300 }],
+  },
+}
+
+const user = userEvent.setup()
+
+describe('CollapsibleTextContent', () => {
+  beforeAll(() => {
+    jest.useFakeTimers()
+  })
+
+  afterAll(() => {
+    jest.useRealTimers()
+  })
+
+  it('should not display all text when text is not expanded', async () => {
+    render(<CollapsibleText numberOfLines={NUMBER_OF_LINES}>{[TEXT]}</CollapsibleText>)
```

Comment:
pourquoi sur certains tests on passe le texte dans un array ?

```suggestion
    render(<CollapsibleText numberOfLines={NUMBER_OF_LINES}>{TEXT}</CollapsibleText>)
```